pkg/apis/core/types.go

```diff
@@ -1598,6 +1601,61 @@ type CSIPersistentVolumeSource struct {
 	NodePublishSecretRef *SecretReference
 }
 
+// Represents a source location of a volume to mount, managed by an external CSI driver
+type CSIVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	// Required.
+	Driver string
+
+	// VolumeHandle is the unique volume name returned by the CSI volume
+	// plugin’s CreateVolume to refer to the volume on all subsequent calls.
+	// If not provided, that handle will be auto-generated.
+	//
+	// +optional
+	VolumeHandle *string
+
+	// Optional: The value to pass to ControllerPublishVolumeRequest.
```

Comment:
non-consistent formatting. I think optional is already captured with the `+optional`  comment line.
Seems like the spacing for these comments is non-consistent too. `VolumeHandle` has space between description and optional, `ReadOnly` does not. I prefer no space.

Also I believe each comment about a type should start with the name of type: for example `Readonly is the value...`